cardano-crypto-class/src/Cardano/Crypto/EllipticCurve/BLS12_381/Internal.hs

```diff
@@ -428,6 +505,26 @@ withNewScalar_ = fmap fst . withNewScalar
 withNewScalar' :: (ScalarPtr -> IO a) -> IO Scalar
 withNewScalar' = fmap snd . withNewScalar
 
+withScalarArray :: [Scalar] -> (ScalarArrayPtr -> IO a) -> IO a
+withScalarArray scalars go = do
+  let numScalars = length scalars
+      sizeReference = sizeOf (undefined :: Ptr ())
+  -- Allocate space for the scalars and a null terminator
```

Comment:
Why the null terminator? AFAICT, all code that consumes this API uses an explicit length to determine the end of the array; the null terminator is never used.

Reply:
Initially, I had it without, but the C code does use it (I discovered this by looking at the Rust implementation of the blst bindings [here](https://github.com/supranational/blst/blob/cbc7e166a10d7286b91a3a7bea341e708962db13/bindings/rust/src/pippenger.rs#L144)).

And to test it without, I ran [this](https://github.com/IntersectMBO/cardano-base/pull/514/commits/677279cc5f038ac548379d25f6fd722effe05df5) CI test, which [failed](https://github.com/IntersectMBO/cardano-base/actions/runs/14728349692/job/41336359225) on all windows machines.

Reply:
And to add to that, there are more things that are implicit in for the blst lib that I discovered. For example, for windows builds, given one pair for an unpiut (so one point + scalar), the msm call will also fail.

And to test it, see [this](https://github.com/IntersectMBO/cardano-base/pull/514/commits/de8c2558c9c143b7b52663a949d46dac1f0fb624) and [this](https://github.com/IntersectMBO/cardano-base/actions/runs/15020090010/job/42206982810) failure on all windows tests.

Reply:
Thank you for digging into this! Ill clean up the commits after the weekend and bump the minor version + add to the changelog. Thanks for the help and the reviews!